sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/LocalCryptographyClient.cs

```diff
@@ -0,0 +1,609 @@
+    /// </summary>
+    public class LocalCryptographyClient : IKeyEncryptionKey
+    {
+        private readonly JsonWebKey _jsonWebKey;
+        private readonly ICryptographyProvider _provider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LocalCryptographyClient"/> class.
+        /// </summary>
+        /// <param name="jsonWebKey">A <see cref="JsonWebKey"/> used for cryptographic operations.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="jsonWebKey"/> is null.</exception>
+        /// <exception cref="NotSupportedException">The <paramref name="jsonWebKey"/> has a <see cref="JsonWebKey.KeyType"/> that is not supported.</exception>
+#pragma warning disable AZC0007 // DO provide a minimal constructor that takes only the parameters required to connect to the service.
+        public LocalCryptographyClient(JsonWebKey jsonWebKey)
+        {
+            _jsonWebKey = jsonWebKey ?? throw new ArgumentNullException(nameof(jsonWebKey));
+            _provider = LocalCryptographyProviderFactory.Create(jsonWebKey, null) ?? throw new NotSupportedException(@$"Key type ""{jsonWebKey.KeyType}"" is not supported");
+        }
+#pragma warning restore AZC0007 // DO provide a minimal constructor that takes only the parameters required to connect to the service.
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LocalCryptographyClient"/> class for mocking.
+        /// </summary>
+        protected LocalCryptographyClient()
+        {
+        }
+
+        /// <summary>
+        /// Gets the <see cref="JsonWebKey.Id"/> of the key used to perform local cryptographic operations for the client.
+        /// </summary>
+        public string KeyId => _jsonWebKey.Id;
+
+        /// <summary>
+        /// Encrypts the specified plain text.
+        /// </summary>
+        /// <param name="algorithm">The <see cref="EncryptionAlgorithm"/> to use.</param>
+        /// <param name="plaintext">The data to encrypt.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> to cancel the operation.</param>
+        /// <returns>
+        /// The result of the encrypt operation. The returned <see cref="EncryptResult"/> contains the encrypted data
+        /// along with all other information needed to decrypt it. This information should be stored with the encrypted data.
+        /// </returns>
+        /// <exception cref="ArgumentException">The specified <paramref name="algorithm"/> does not match the key corresponding to the key identifier.</exception>
+        /// <exception cref="InvalidOperationException">The key is invalid for the current operation.</exception>
+        /// <exception cref="NotSupportedException">The operation is not supported with the specified key.</exception>
+        public virtual async Task<EncryptResult> EncryptAsync(EncryptionAlgorithm algorithm, byte[] plaintext, CancellationToken cancellationToken = default)
```

Comment:
nit: Couldn't all of these also throw `CryptographicException` as well? should we document this?

Reply:
Yes. Sorry, I meant to add those both here and in `CryptographyClient` as well.